### PR TITLE
Add support for authentication properties defined in autorest/modelerfour

### PR DESCRIPTION
**Pull request by ShivangiReja**
Fixes: https://github.com/Azure/autorest.csharp/issues/1106

Blocked by https://github.com/Azure/autorest/pull/4018

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1000,7 +1000,7 @@ input-file: "swagger-document"
 
 ```yaml
 # autorest-core version
-version: 3.1.3
+version: "https://tinyurl.com/yfhyccns"
```
Comment (ShivangiReja) [inline readme.md:1003]: Update version after https://github.com/Azure/autorest/pull/4018 PR merges

File: src/AutoRest.CSharp/readme.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 ## Configuration
 ```yaml
 use-extension:
-  "@autorest/modelerfour": "4.17.2"
+  "@autorest/modelerfour": "https://tinyurl.com/yfratqhg"
```
Comment (ShivangiReja) [inline src/AutoRest.CSharp/readme.md:7]: Update version after https://github.com/Azure/autorest/pull/4018 PR merges

File: src/AutoRest.CodeModel/Program.cs [hunk 3]
```diff
@@ -13,13 +13,13 @@ namespace AutoRest.CodeModel
...
     {
-        private const string Path = "AutoRest.CSharp/Input/Generated";
+        private const string Path = "AutoRest.CSharp/Common/Input/Generated";
         private static readonly string Namespace = "AutoRest.CSharp.Input";
 
         private static void Main()
         {
             using var webClient = new WebClient();
-            webClient.DownloadFile(@"https://raw.githubusercontent.com/Azure/autorest/master/packages/libs/codemodel/.resources/all-in-one/json/code-model.json", "code-model.json");
+            webClient.DownloadFile(@"https://raw.githubusercontent.com/Azure/autorest/b5950b43bb258f1a33e39d5506c8a449ddaf5e89/packages/libs/codemodel/.resources/all-in-one/json/code-model.json", "code-model.json");
```
Comment (ShivangiReja) [inline src/AutoRest.CodeModel/Program.cs:22]: Update link after https://github.com/Azure/autorest/pull/4018 PR merges.

**Comment by chamons** [comment]
This was blocked on https://github.com/Azure/autorest/pull/4018 landing in autorest core.

**Comment by timotheeguerin** [comment]
Make sure to also update the LLC docs where you referenced the old config https://github.com/Azure/autorest.csharp/blob/feature/v3/docs/protocol_method_client.md

**Comment by chamons** [comment]
Starting work to bring this back to life. A number of merge issues to resolve before review.

**Comment by chamons** [comment]
I have it doing Generate.ps1 locally with no changes after fixing a bunch of project files.

File: eng/Generate.ps1 [hunk 4]
```diff
@@ -84,7 +84,7 @@ if (!($Exclude -contains "TestServer"))
     }
 }
 
-$llcArgs = "--low-level-client=true --credential-types=AzureKeyCredential --credential-header-name=Fake-Subscription-Key"
+$llcArgs = "--low-level-client=true --security=AzureKey --security-header-name=Fake-Subscription-Key"
```
Comment (pakrym) [inline eng/Generate.ps1:87]: Don't we support no-cred LLC now?

Comment (chamons) [reply eng/Generate.ps1:87]: Yep, I could just drop this to be honest. I just figured the key credential was more of the "happy path" than the uncommon no-auth.

File: src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs [hunk 5]
```diff
@@ -275,7 +280,7 @@ private void WriteClientCtors(CodeWriter writer, LowLevelRestClient client, Buil
             {
                 WriteConstructor(writer, client, CredentialKind.Token, context);
             }
-            if (context.Configuration.CredentialTypes.Length == 0)
+            if (!context.CodeModel.Security.Schemes.Any())
```
Comment (pakrym) [inline src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs:283]: Can we use the same foreach pattern as dataplane uses here?

Comment (chamons) [reply src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs:283]: Maybe, but there is a difference - DataPlaneClientWriter doesn't have a fallback "if none of these, do this other one". Here we want to only generate the no-auth constructor if no other one was created. 

I could use a foreach and a bool, but that felt worse to me.

Comment (pakrym) [reply src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs:283]: Yeah, but LLC has 3 patterns now: foreach loops, bools, and enum. Seems like too many ways to talk about the same thing.

Comment (chamons) [reply src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs:283]: Fair enough. Let me look

**Comment by pakrym** [comment]
@ShivangiReja please review.

**Comment by ShivangiReja** [comment]
LGTM, @chamons can't approve because it's my own pr. Thank you for your work!

**Review by pakrym — APPROVED** [review]

**Pull request merged**